cirq/contrib/noise_models/noise_models.py

```diff
@@ -0,0 +1,152 @@
+        else:
+            if is_measurements is None:
+                is_measurements = False
+            elif not is_measurements:
+                # Check that all are *not* measurements
+                pass
+            else:
+                raise ValueError("Moment must be all measurements "
+                                 "or all operations.")
+    return is_measurements
+
+
+class DepolarizingNoiseModel(devices.NoiseModel):
+    """Applies depolarizing noise to each qubit individually at the end of
+    every moment.
+
+    If a circuit contains measurements, they be in moments that don't
+    also contain gates.
+
+    Args:
+        depol_prob: Depolarizing probability.
+    """
+
+    def __init__(self, depol_prob: float):
+        value.validate_probability(depol_prob, 'depol prob')
+        self.qubit_noise_gate = ops.DepolarizingChannel(depol_prob)
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _moment_is_measurements(moment):
+            return moment
+
+        return [
+            moment,
+            ops.Moment(self.qubit_noise_gate(q) for q in system_qubits)
+        ]
+
+
+class DepolarizingWithReadoutNoiseModel(devices.NoiseModel):
+    """DepolarizingNoiseModel with probabilistic bit flips preceding
+    measurement.
+
+    This simulates readout error.
+
+    If a circuit contains measurements, they be in moments that don't
+    also contain gates.
+
+    Args:
+        depol_prob: Depolarizing probability.
+        bitflip_prob: Probability of a bit-flip during measurement.
+    """
+
+    def __init__(self, depol_prob: float, bitflip_prob: float):
+        value.validate_probability(depol_prob, 'depol prob')
+        value.validate_probability(bitflip_prob, 'bitflip prob')
+        self.qubit_noise_gate = ops.DepolarizingChannel(depol_prob)
+        self.readout_noise_gate = ops.BitFlipChannel(bitflip_prob)
+
+    def noisy_moment(self, moment: 'cirq.Moment',
+                     system_qubits: Sequence['cirq.Qid']):
+        if _moment_is_measurements(moment):
+            return [
+                ops.Moment(self.readout_noise_gate(q) for q in system_qubits),
+                moment,
+            ]
+        return [
+            moment,
+            ops.Moment(self.qubit_noise_gate(q) for q in system_qubits),
+        ]
+
+
+class DepolarizingWithDampedReadoutNoiseModel(devices.NoiseModel):
+    """DepolarizingWithReadoutNoiseModel with T1 decay preceding
+    measurement.
+
+    This simulates asymmetric readout error. The noise is structured
+    so the T1 decay is applied, then the readout bitflip, then measurement.
+
+    If a circuit contains measurements, they be in moments that don't
+    also contain gates.
+
+    Args:
+        depol_prob: Depolarizing probability.
+        bitflip_prob: Probability of a bit-flip during measurement.
+        decay_prob: Probability of T1 decay during measurement.
```

Comment:
How does this probability combine with the bitflip probability? They don't commute with each other.

For example, if I set both probabilities to 1, will the measurement always report true or will it always report false?

Reply:
This is addressed a bit in the body of the docstring

>    This simulates asymmetric readout error. The noise is structured
>    so the T1 decay is applied, then the readout bitflip, then measurement.

I think this is more physical. I imagine the bitflip is more of a classification error and the t1 decay is something physically going wrong in measurement. If that matches reality, then the current order makes sense.

Reply:
added to Args as well